SwrveSDK/SDK/Track/Swrve.m

```diff
@@ -814,10 +814,45 @@ -(id) initWithAppID:(int)swrveAppID apiKey:(NSString*)swrveAPIKey config:(SwrveC
     }
 
     [self sendQueuedEvents];
-
+    
+    // legacy from 4.7
+    [self migrateFileProtection];
+    
     return self;
 }
 
+- (BOOL) isProtectedItemAtPath:(NSString *)path {
+    BOOL            result                      = YES;
+    NSDictionary    *attributes                 = nil;
+    NSString        *protectionAttributeValue   = nil;
+    NSFileManager   *fileManager                = nil;
+    NSError         *error                      = nil;
+    
+    fileManager = [[NSFileManager alloc] init];
+    attributes = [fileManager attributesOfItemAtPath:path error:&error];
+    if (attributes != nil){
+        protectionAttributeValue = [attributes valueForKey:NSFileProtectionKey];
+        if ((protectionAttributeValue == nil) || [protectionAttributeValue isEqualToString:NSFileProtectionNone]){
+            result = NO;
+        }
+    } else {
+        DebugLog(@"There was an issue finding the level of file protection for : %@  \nError: %@" , path , error);
+    }
+    return result;
+}
+
+
+- (void) migrateFileProtection {
+
+    if([self isProtectedItemAtPath:[SwrveFileManagement applicationSupportPath]]){
```

Comment:
So logic would be to store current version number, and in the migrations, check if current version number is less than it should be --> if it is, then run migrations, if there are any, and then increment the current version number.